notebook/tests/selenium/utils.py

```diff
@@ -382,3 +386,77 @@ def trigger_keystrokes(browser, *keys):
             ac.perform()
         else:              # single key stroke. Check if modifier eg. "up"
             browser.send_keys(getattr(Keys, keys[0].upper(), keys[0]))
+
+def validate_dualmode_state(notebook, mode, index):
+    '''Validate the entire dual mode state of the notebook.  
+    Make sure no more than one cell is selected, focused, in edit mode, etc...
+
+    '''
+    def is_only_cell_edit(index):
+        JS = 'return Jupyter.notebook.get_cells().map(function(c) {return c.mode;})'
+        cells_mode = notebook.browser.execute_script(JS)
+        #None of the cells are in edit mode
+        if(index == None):
```

Comment:
By convention, we check for None using `index is None` rather than `index == None`.

This can matter in some situations, because objects can override the `==` operator and do strange things. But it's also just a common convention for Python code.